Signal/src/call/CallService.swift

```diff
@@ -494,16 +488,25 @@ protocol CallServiceObserver: class {
 
         guard self.call == nil else {
             // TODO on iOS10+ we can use CallKit to swap calls rather than just returning busy immediately.
-            Logger.info("\(TAG) receivedCallOffer for thread: \(thread) but we're already in call: \(call!)")
+            Logger.info("\(TAG) receivedCallOffer: \(newCall.debugString) but we're already in call: \(call!.debugString)")
 
             handleLocalBusyCall(newCall, thread: thread)
 
+            if self.call?.remotePhoneNumber == newCall.remotePhoneNumber {
+                // If we're receiving a new call offer from the user we think we have a call with, terminate
+                // our current call to get back to a known good state.  If they call back, we'll be ready.
+                // 
+                // TODO: Auto-accept this incoming call if our current call was either a) outgoing or 
+                // b) ever connected.  There will be a bit of complexity around making sure that two
+                // parties that call each other at the same time end up connected.
+                terminateCall()
```

Comment:
I haven't run this - what does this user see locally when they terminate their call? 

Presumably if both users are calling each other at the same time, they'll *both* hit this logic, and *both* terminate their local call?